ydb/src/session.rs

```diff
@@ -1,252 +1,272 @@
+                tx_id,
+                operation_params: self.timeouts.operation_params(),
+                collect_stats: DEFAULT_COLLECT_STAT_MODE,
+            })
+            .await;
+        self.handle_raw_result(res)?;
+        Ok(())
+    }
+
+    pub(crate) async fn execute_schema_query(&mut self, query: String) -> YdbResult<()> {
+        let res = self
+            .get_table_client()
+            .await?
+            .execute_scheme_query(RawExecuteSchemeQueryRequest {
+                session_id: self.id.clone(),
+                yql_text: query,
+                operation_params: self.timeouts.operation_params(),
+            })
+            .await;
+        self.handle_raw_result(res)?;
+        Ok(())
+    }
+
+    #[tracing::instrument(skip(self, req), fields(req_number=req_number()))]
+    pub(crate) async fn execute_data_query(
+        &mut self,
+        mut req: RawExecuteDataQueryRequest,
+        error_on_truncated: bool,
+    ) -> YdbResult<QueryResult> {
+        req.session_id.clone_from(&self.id);
+        req.operation_params = self.timeouts.operation_params();
+
+        trace!(
+            "request: {}",
+            ensure_len_string(serde_json::to_string(&req)?)
+        );
+
+        let res = self.get_table_client().await?.execute_data_query(req).await;
+        let res = self.handle_raw_result(res)?;
+        trace!(
+            "result: {}",
+            ensure_len_string(serde_json::to_string(&res)?)
+        );
+        if error_on_truncated {
+            return Err(YdbError::from_str("result of query was truncated"))
+        }
+        QueryResult::from_raw_result( error_on_truncated, res)
+    }
+
+    #[tracing::instrument(skip(self, query), fields(req_number=req_number()))]
+    pub async fn execute_scan_query(&mut self, query: Query) -> YdbResult<StreamResult> {
+        let req = ExecuteScanQueryRequest {
+            query: Some(query.query_to_proto()),
+            parameters: query.params_to_proto()?,
+            mode: execute_scan_query_request::Mode::Exec as i32,
+            ..ExecuteScanQueryRequest::default()
+        };
+        debug!("request: {}", crate::trace_helpers::ensure_len_string(serde_json::to_string(&req)?));
+        let mut channel = self.get_channel().await?;
+        let resp = channel.stream_execute_scan_query(req).await?;
+        let stream = resp.into_inner();
+        Ok(StreamResult { results: stream })
+    }
+
+    pub(crate) async fn rollback_transaction(&mut self, tx_id: String) -> YdbResult<()> {
+        let mut table = self.get_table_client().await?;
+        let res = table
+            .rollback_transaction(RawRollbackTransactionRequest {
+                session_id: self.id.clone(),
+                tx_id,
+                operation_params: self.timeouts.operation_params(),
+            })
+            .await;
+
+        self.handle_raw_result(res)
+    }
+
+    #[allow(dead_code)]
+    pub(crate) async fn copy_table(

```

Comment:
copy_table must be public function (without crate restriction) - for use from outside of the crate.